### ISSUE TITLE
Try to create a stable ABI

### ISSUE THREAD

**Issue by nahuelwexd** [nahuelwexd/Vadi#6]
Dunno how to do this on Vala, so I would need to investigate

**Comment by sdv43** [comment]
Hi! Are you still planning to develop the library? Or is there no interest?

**Comment by nyx-lyb3ra** [comment]
@sdv43 Yes! In fact I have several changes in local, I just haven't had time to finish them and push them.

**Comment by sdv43** [comment]
Good. I am planning to use this library in https://github.com/sdv43/whaler. I tried to use it, but vapi files are not visible in the system after installation. Have you tried using it from another project as a library?

**Comment by nyx-lyb3ra** [comment]
> but vapi files are not visible in the system after installation

Weird thing... have you checked the installation logs? It should say something like: `Installing lib/vadi-0.0.vapi to /usr/share/vala/vapi`. Also, make sure that you're installing with a `/usr` prefix and not `/usr/local`. Meson tends to default to `/usr/local`.

> Have you tried using it from another project as a library?

Yes, but only as a meson subproject, not as a global dependency

**Comment by sdv43** [comment]
Yes, you are right. I've installed it incorrectly. 

It would be cool to see the first version in the near future. What kind of help do you need?

**Comment by nyx-lyb3ra** [comment]
> What kind of help do you need?

If you can give me a list of the use cases you need in a dependency injection library, that would be great :)

**Comment by sdv43** [comment]
As I understood from the [tests](https://github.com/nahuelwexd/Vadi/blob/main/tests/test-container.vala), all the main use cases have already been implemented.

Specifically, in my case, I need a library to mock objects in tests

**Comment by nyx-lyb3ra** [comment]
> As I understood from the [tests](https://github.com/nahuelwexd/Vadi/blob/main/tests/test-container.vala), all the main use cases have already been implemented.

Great then :)

> Specifically, in my case, I need a library to mock objects in tests

I think that belongs to the scope of a unit testing library. Geary has its own one called [ValaUnit](https://gitlab.gnome.org/GNOME/geary/-/tree/mainline/subprojects/vala-unit), based on GTest, maybe you should take a look at it.
